datastore/shared/postgresql_backend/schema.sql

```diff
@@ -11,54 +11,101 @@
 DO $$
 BEGIN
     IF NOT EXISTS (SELECT 1 FROM pg_type WHERE typname = 'event_type') THEN
-        CREATE TYPE event_type AS ENUM ('create', 'update', 'delete', 'deletefields', 'restore', 'noop');
+        CREATE TYPE event_type AS ENUM ('create', 'update', 'delete', 'deletefields', 'listfields', 'restore');
     ELSE
         RAISE NOTICE 'type "event_type" already exists, skipping';
     END IF;
 END$$;
 
 
 CREATE TABLE IF NOT EXISTS positions (
-    position SERIAL PRIMARY KEY,
+    position INTEGER PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
     timestamp TIMESTAMPTZ,
     user_id INTEGER NOT NULL,
-    information JSON
+    information JSON,
+    migration_index INTEGER NOT NULL
 );
 
+-- Keep this schema in sync with `migration_events`, see below!
+-- This includes indices, ...
 CREATE TABLE IF NOT EXISTS events (
-    id BIGSERIAL PRIMARY KEY,
+    id BIGINT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
     position INTEGER REFERENCES positions(position) ON DELETE CASCADE,
     fqid VARCHAR(48) NOT NULL,
     type event_type NOT NULL,
-    data JSONB
+    data JSONB,
+    weight INTEGER NOT NULL
 );
-CREATE INDEX IF NOT EXISTS events_fqid_idx ON events (fqid);
+CREATE INDEX IF NOT EXISTS events_1_fqid_idx ON events (fqid);
 -- TODO: create index for data->>meeting_id for collectionfieldlocks
 
-CREATE TABLE IF NOT EXISTS models_lookup (
-    fqid VARCHAR(48) PRIMARY KEY,
-    deleted BOOLEAN NOT NULL
-);
-
+-- For the `reserve_ids` feature
 CREATE TABLE IF NOT EXISTS id_sequences (
     collection VARCHAR(32) PRIMARY KEY,
     id INTEGER NOT NULL
 );
 
+-- Helper tables
 CREATE TABLE IF NOT EXISTS collectionfields (
-    id BIGSERIAL PRIMARY KEY,
+    id BIGINT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
     collectionfield VARCHAR(255) UNIQUE NOT NULL,
     position INTEGER NOT NULL
 );
 CREATE INDEX IF NOT EXISTS collectionfields_collectionfield_idx on collectionfields (collectionfield);
 
 CREATE TABLE IF NOT EXISTS events_to_collectionfields (
-    event_id BIGINT REFERENCES events(id) ON UPDATE CASCADE ON DELETE CASCADE,
-    collectionfield_id BIGINT REFERENCES collectionfields(id) ON UPDATE CASCADE ON DELETE CASCADE,
+    event_id BIGINT, -- no reference to events(id) here since it is swapped with migration_events on
```

Comment:
> I think you can deactivate foreign key checks temporarily

But I do not know how postgres saves the referenced table. If it is by name it will be fine. If there is an internal "pointer", this won't work. To not build upon postgres implementation details, I suggest leaving it as-is.

Reply:
Ok, sure